src/main/scala/is/hail/io/plink/PlinkLoader.scala

```diff
@@ -100,15 +96,21 @@ object PlinkLoader {
               case numericRegex() => fatal(s"Invalid case-control phenotype: `$pheno'. Control is `1', case is `2', missing is `0', `-9', `${ ffConfig.missingValue }', or non-numeric.")
               case _ => null
             }
+        idBuilder += kid
+        structBuilder += Annotation(fam1, dad1, mom1, isFemale1, pheno1)
+      }
+    }
 
-        (kid, Annotation(fam1, dad1, mom1, isFemale1, pheno1))
-      }.value).toIndexedSeq
+    val (sampleIds, duplicates) = mangle(idBuilder.result(), "D" * _)
+    if (duplicates.nonEmpty) {
+      warn(s"Found ${ duplicates.length } duplicate ${ plural(duplicates.length, "sample ID") }. Mangled IDs follows:\n  @1",
+        duplicates.map { case (pre, post) => s"'$pre' -> '$post'"}.truncatable("\n  "))
```

Comment:
I agree. One solution is to only use the unique IDs internally and keep track of the mapping between IDs. This would mean we would need to change `annotate_samples_*` to support this.